Source/DafnyStandardLibraries/src/Std/Termination.dfy

```diff
@@ -0,0 +1,79 @@
+
+
+module Std.Termination {
+
+  datatype ClauseTail = More(next: TerminationMetric) | Top
+
+  datatype TerminationMetric = TerminationMetric(first: TMValue, rest: ClauseTail) {
+    predicate DecreasesTo(other: TerminationMetric) {
+      if first == other.first then
+        match (rest, other.rest) {
+          case (_, Top) => false
+          case (Top, More(_)) => true
+          case (More(next), More(otherNext)) => next.DecreasesTo(otherNext)
+        }
+      else
+        first.DecreasesTo(other.first)
+    }
+
+    ghost function {:axiom} Ordinal(): ORDINAL
```

Comment:
Given how few operations we have defined on `ORDINAL`, this is probably no different than `nat` in our current state.

Reply:
Agreed, and I'm going to clean this up a bit so we don't claim to support anything other than nat for now, but at least have this concept in place in order to define the signature for `GenericAction`